pkg/destinations/destinations.go

```diff
@@ -83,17 +93,21 @@ func (m *DestinationManager) Destination(ctx context.Context, databaseID int64)
 		defer m.mux.Unlock(databaseID)
 
 		var dest Destination
+		var creds Creds
 
 		dest, ok := m.pool[databaseID]
 		if ok {
 			return dest, nil
 		}
 
-		creds, err := m.storage.Database.GetDestinationCredentials(ctx, databaseID)
+		jsonCreds, err := m.storage.Vault.GetCredential(strconv.Itoa(int(databaseID)))
```

Comment:
We only need to store the value of `Settings` in the vault. Everything else we should fetch from the database. Perhaps the [GetDestinationCredentials()](https://github.com/scratchdata/scratchdata/blob/main/pkg/storage/database/database.go#L17) function should be renamed to `GetDestination()` and return all the same values except for Settings.

Reply:
@poundifdef I need some more clarity on this point. Currently the memory vault has this param : `destinations []config.Destination`. Do you want me to remove Settings from Destination struct ?

If yes, then what should be the new param to the function ?